Sources/MapboxDirections/RouteOptions.swift

```diff
@@ -92,6 +92,11 @@ open class RouteOptions: DirectionsOptions {
             self.maximumWeight = Measurement(value: doubleValue, unit: UnitMass.metricTons)
         }
         
+        if let mappedValue = mappedQueryItems[CodingKeys.layers.stringValue] {
+            zip(waypoints, mappedValue.components(separatedBy: ";")).forEach {
+                $0.layer = Int($1) ?? nil
+            }
+        }
```

Comment:
This differs a bit from how other per-waypoint properties like `separatesLegs` (aka `waypoints` in the API request) are implemented:

https://github.com/mapbox/mapbox-directions-swift/blob/53f2d24edb78db8ea8cb12147be521b05d976a16/Sources/MapboxDirections/DirectionsOptions.swift#L172-L179

What you’ve got makes a bit more sense, though, so we should double-check that the `separatesLegs` code is correct.